test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -4334,6 +4476,274 @@ def test_cgroup_default_config(self):
         self.server.delete(id=jid, wait=True)
         self.assertFalse(self.is_dir(ehjd1, ehost1), "job cpuset dir found")
 
+    def test_cgroup_cgswap(self, vnode_per_numa_node=False):
+        if not self.mem:
+            self.skipTest('Test requires memory subystem mounted')
+        if not self.swapctl:
+            self.skipTest('Test requires memsw accounting enabled')
+        self.server.remove_resource('cgswap')
+        self.server.add_resource('cgswap', 'size', 'nh')
+        self.scheduler.add_resource('cgswap')
+        events = ['execjob_begin', 'execjob_launch', 'execjob_attach',
+                  'execjob_epilogue', 'execjob_end', 'exechost_startup',
+                  'exechost_periodic', 'execjob_resize', 'execjob_abort',
+                  'queuejob', 'modifyjob']
+        # Enable the cgroups hook new events
+        conf = {'enabled': 'True', 'freq': 10, 'event': events}
+        self.server.manager(MGR_CMD_SET, HOOK, conf, self.hook_name)
+
+        self.load_config(self.cfg15
+                         % ('true' if vnode_per_numa_node else 'false'))
+        # Restart mom for changes made by cgroups hook to take effect
+        begin = time.time()
+        # sleep 2s to ensure log matching will not catch older log lines
+        time.sleep(2)
```

Comment:
No, and yes.
 
We need a time that is guaranteed to be before the action so that we know from when to search in the logs, but we also need to avoid catching the results of earlier actions; the way we do that is to take time, then not do anything for 2 seconds. 

1 second is not enough for reasons of rounding -- time.time is a float but times in the logs are integers, and very small rounding differences might (rarely) make PTL and the daemons come to a different conslusion.

Reply:
In the interest of readability:
-sleep 1 (to ensure that any 'old' log line no longer matches the "begin" timestamp)
-begin = time.time()
-sleep 1 (to ensure that rounding errors do not make us "miss" a line caused by the next action
-action
-log inspection

Reply:
I have adapted all tests to use this newer flow. And yes, those time.sleep(1) are pretty minimal -- we don't have microsecond logging enabled everywhere ;-). And unfortunately on a wll-know test harness I have proof by existence that small time differences in recorded log times between PTL and the daemons can wreak havoc if you try not to sleep, and the resolution of the log lines is 1s.